gherkin/php/src/Token.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Cucumber\Gherkin;
+
+use Cucumber\Gherkin\Parser\TokenType;
+
+final class Token
+{
+    public ?TokenType $matchedType = null;
+    public ?string $matchedKeyword = null;
+    public ?string $matchedText = null;
+    public ?int $matchedIndent = null;
+    public ?GherkinDialect $matchedGherkinDialect = null;
+
+    /** @var list<GherkinLineSpan> */
+    public ?array $matchedItems = null;
+
+    public function __construct(
+        public readonly ?GherkinLine $line,
+        /**
+         * Public because the token matcher adds indent to columns
+         */
+        public Location $location,
+    ) {
+    }
+
+    /**
+     * @psalm-assert-if-true null $this->line
+     */
+    public function isEof(): bool
+    {
+        return is_null($this->line);
+    }
+
+    public function detach(): void
```

Comment:
Find out why this is a no-op

Reply:
From Slack, we aren't sure what it does and if it can be removed, but @gasparnagy may know!

Reply:
I don't remember fully, but it does not seem to be in use. I remember that I did a token/line implementation that did nor require to split the feature file text into multiple strings (memory optimization) and for some reason it was important to that implementation to know when the token is not used anymore. But I can't really imagine why was that needed and my optimized implementation seem not to be used anymore, probably deleted. 

I guess the parser generator will generate this call and tells to token when it is not in use anymore, but feel free to not do anything there.

Reply:
Thanks, removed it! We can always add back in :)